### PR TITLE
Add Supplier feature with CLI parsing support

### PR DESCRIPTION
**Pull request by yikyak02**
- Implement Supplier class to represent suppliers in the address book
  - Includes fields for name, phone, email, address, dietary preference, ingredients supplied, and tags
  - Add methods for managing open supply orders and ingredients supplied

- Add AddSupplierCommand class to handle the addition of suppliers
  - Executes the logic for adding a new supplier to the address book
  - Includes necessary validation and duplicate supplier checks

- Implement AddSupplierCommandParser class to parse supplier addition commands from the CLI
  - Parses input fields like name, phone, email, address, preference, and ingredients supplied
  - Uses the new IngredientsParser method to parse ingredients supplied into a structured data type

- Refactore the logic for parsing ingredients supplied in ParserUtil
  - Create parseIngredients method to handle the parsing of ingredients supplied from a comma-separated string input

### CODE REVIEW

**Review by ryenl — APPROVED** [review]
Seems like you need to work on how to handle inputting ingredients sold when adding a supplier, otherwise LGTM!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -81,6 +86,31 @@ public static Address parseAddress(String address) throws ParseException {
         }
         return new Address(trimmedAddress);
     }
+    /**
+     * Parses a {@code String ingredientsSuppliedString} into an {@code Ingredients} object.
+     * Splits the string by commas and trims whitespaces.
+     *
+     * @throws ParseException if the given {@code ingredientsSuppliedString} is invalid.
+     */
+    public static Ingredients parseIngredients(String ingredientsSuppliedString) throws ParseException {
```
Comment (calebchongsj) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:95]: Nice to see that you have parsed the ingredients for the supplier.

File: src/main/java/seedu/address/logic/parser/AddSupplierCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_PREFERENCE, PREFIX_INGREDIENTS_SUPPLIED, PREFIX_TAG);
+
+        if (!arePrefixesPresent(argMultimap, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS, PREFIX_PREFERENCE, PREFIX_INGREDIENTS_SUPPLIED)) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddSupplierCommand.MESSAGE_USAGE));
+        }
+
+        Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
+        Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
+        Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+        Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
```
Comment (calebchongsj) [inline src/main/java/seedu/address/logic/parser/AddSupplierCommandParser.java:34]: Great job with adding the relevant attributes for the supplier.

**Review by calebchongsj — APPROVED** [review]
LGTM. Great job Lucas

**Pull request merged**